File: tests/unit-tests/file_tests/non-deterministic/clock_gettime_simple.c
#include <time.h>
#include <stdio.h>

int main() {
    struct timespec tp;

    // get CLOCK_REALTIME's time
    if (clock_gettime(CLOCK_REALTIME, &tp) == -1) {
        perror("clock_gettime failed");
        return 1;
    }

    //print the result
    printf("Current time: %lld seconds and %ld nanoseconds\n", tp.tv_sec, tp.tv_nsec);

    return 0;
}

File: tests/unit-tests/file_tests_rust/non-deterministic/clock_gettime_simple.rs
use std::time::{SystemTime, UNIX_EPOCH};

fn main() {
    // Get current time using SystemTime
    match SystemTime::now().duration_since(UNIX_EPOCH) {
        Ok(duration) => {
            let seconds = duration.as_secs();
            let nanoseconds = duration.subsec_nanos();
            println!("Current time: {} seconds and {} nanoseconds", seconds, nanoseconds);
        }
        Err(e) => {
            eprintln!("clock_gettime failed: {}", e);
        }
    }
} 